test/test_tensordict.py

```diff
@@ -80,6 +81,23 @@ def test_stack(device):
     assert (td_reconstruct == td).all()
 
 
+@pytest.mark.parametrize("device", get_available_devices())
+def test_pad(device):
```

Comment:
Let's use `TestTensorDict`

Have a look at the tests from that class: you can test your function for each tensordict subclass.

```python
    def test_pad(self, td_name):
        torch.manual_seed(1)
        td = getattr(self, td_name)
        etc.
```